CLAUDE.md

```diff
@@ -0,0 +1,126 @@
+# CLAUDE.md
+
+This file provides guidance to Claude Code (claude.ai/code) when working with code in this repository.
+
+## Project Overview
+
+torchao is PyTorch's official Architecture Optimization library that accelerates PyTorch models through advanced quantization and sparsification techniques. It provides optimization for weights, gradients, activations, and more for both inference and training with minimal code changes.
+
+## Development Commands
+
+### Installation & Build
+```bash
+# Development install (Python-only mode, fastest for development)
+USE_CPP=0 python setup.py develop
+
+# Full build with C++/CUDA extensions
+python setup.py develop
+
+# Install specific version of ruff for linting
+pip install ruff==0.11.6
+```
+
+### Testing
+```bash
+# Run specific test files
+pytest test/float8/test_base.py
+pytest test/quantization/test_quant_api.py
+pytest test/dtypes/test_affine_quantized.py
+
+# Run comprehensive float8 tests
+./test/float8/test_everything.sh
+
+# Run all tutorials
+./tutorials/run_all.sh
+```
+
+### Linting & Formatting
+```bash
+# Install pre-commit hooks (one-time setup)
+pre-commit install
+
+# Run all pre-commit checks
+pre-commit run --all-files
+
+# Run pre-commit on staged files only
+pre-commit run
+```
+
+## Architecture Overview
+
+### Core Components
+
+**torchao/quantization/** - Primary quantization APIs
+- `quant_api.py` - Main `quantize_()` function for one-line model quantization
+- `autoquant.py` - Automatic quantization selection
+- Weight-only quantization (INT4/INT8), dynamic quantization, QAT support
+
+**torchao/dtypes/** - Custom tensor subclasses with layout and dispatch registration
+- `AffineQuantizedTensor` - Base quantized tensor class
+- `nf4tensor.py` - NF4 (4-bit normal float) implementation for QLoRA
+- `uintx/floatx/` - Unsigned integer and floating-point quantized tensors
+
+**torchao/float8/** - High-performance float8 training
+- Delivers up to 1.5x speedup on 512 GPU clusters
```

Comment:
Compared to what?

Reply:
fwiw I just ran CC's `/setup` command to generate  but yeah this likely doens't help the model